src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1893,6 +1893,74 @@ bool LibraryCallKit::inline_vector_blend() {
   return true;
 }
 
+// public static
+// <V extends Vector<E>,
+//  E>
+// V slice(Class<? extends V> vectorClass, Class<E> elementType, int vlen,
+//         V v1, V v2, int origin,
+//         VectorSliceOp<V> defaultImpl)
+bool LibraryCallKit::inline_vector_slice() {
+  const TypeInstPtr* vector_klass = gvn().type(argument(0))->isa_instptr();
+  const TypeInstPtr* elem_klass   = gvn().type(argument(1))->isa_instptr();
+  const TypeInt*     vlen         = gvn().type(argument(2))->isa_int();
+  const TypeInt*     origin       = gvn().type(argument(5))->isa_int();
+
+  if (vector_klass == NULL || elem_klass == NULL || vlen == NULL || origin == NULL) {
+    return false; // dead code
+  }
+  if (vector_klass->const_oop() == NULL || elem_klass->const_oop() == NULL ||
+      !vlen->is_con() || !origin->is_con()) {
+    if (C->print_intrinsics()) {
+      tty->print_cr("  ** missing constant: vclass=%s etype=%s vlen=%s origin=%s",
+                    NodeClassNames[argument(0)->Opcode()],
+                    NodeClassNames[argument(1)->Opcode()],
+                    NodeClassNames[argument(2)->Opcode()],
+                    NodeClassNames[argument(5)->Opcode()]);
+    }
+    return false; // not enough info for intrinsification
+  }
+  if (!is_klass_initialized(vector_klass)) {
+    if (C->print_intrinsics()) {
+      tty->print_cr("  ** klass argument not initialized");
+    }
+    return false;
+  }
+  ciType* elem_type = elem_klass->const_oop()->as_instance()->java_mirror_type();
+  if (!elem_type->is_primitive_type()) {
+    if (C->print_intrinsics()) {
+      tty->print_cr("  ** not a primitive bt=%d", elem_type->basic_type());
+    }
+    return false; // should be primitive type
+  }
+  BasicType elem_bt = elem_type->basic_type();
+  int num_elem = vlen->get_con();
+
+  if (!arch_supports_vector(Op_VectorSlice, num_elem, elem_bt, VecMaskNotUsed)) {
+    if (C->print_intrinsics()) {
+      tty->print_cr("  ** not supported: arity=2 op=slice vlen=%d etype=%s ismask=notused",
+                    num_elem, type2name(elem_bt));
+    }
+    return false; // not supported
+  }
+  ciKlass* vbox_klass = vector_klass->const_oop()->as_instance()->java_lang_Class_klass();
+  const TypeInstPtr* vbox_type = TypeInstPtr::make_exact(TypePtr::NotNull, vbox_klass);
+
+  Node* v1   = unbox_vector(argument(3), vbox_type, elem_bt, num_elem);
+  Node* v2   = unbox_vector(argument(4), vbox_type, elem_bt, num_elem);
+  Node* o    = argument(5);
+
+  if (v1 == NULL || v2 == NULL) {
+    return false; // operand unboxing failed
+  }
```

Comment:
Suggest to reorder line-1950 and the if-statement in line-1952. And then we doesn't need too more spaces in the variable definition `Node* v1 = unbox_vector(xxx)`. Besides, could we rename variable `o` to `index` or `origin` ? I know you'v used `origin` at the begin, maybe we can rename it to `origin_type`. I see the similari name style in `inline_vector_frombits_coerced`.